nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -248,6 +220,43 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
 
     }
 
+    protected void writeActors(List<String> actors, boolean isFetchActors, JsonGenerator jg) throws IOException {
+        for (String actorId : actors) {
+            if (isFetchActors) {
+                Object actor = fetchActor(actorId);
+                if (actor != null) {
+                    writeEntity(actor, jg);
+                }
+            } else {
+                jg.writeStartObject();
+                jg.writeStringField("id", actorId);
+                jg.writeEndObject();
+            }
+        }
+    }
+
+    protected Object fetchActor(String actorId) {
+        String unprefixedActorId = getUnprefixedActorId(actorId);
+        Object actor = userManager.getPrincipal(unprefixedActorId);
+        if (actor == null) {
+            actor = userManager.getGroup(unprefixedActorId);
+        }
+        return actor;
+    }
+
+    protected String getUnprefixedActorId(String actorId) {
+        String unprefixedActorId = actorId;
```

Comment:
According to https://doc.nuxeo.com/corg/java-code-style/#you-can-reassign-method-parameters we allow method parameters reassignment.